src/server/index.ts

```diff
@@ -10,38 +11,85 @@ type Action = 'rock' | 'paper' | 'scissors' | null;
 
 const playerSelectedAction: { [key: string]: Action } = {}; 
 
-function setRemainingTime(player: User, seconds: number) {
+function setRemainingTime(player: User, seconds: number) 
+{
 	gameServer.setPrivateData(player.id, 'remaining-time', seconds);
 }
 
-function pickAction(user: User, action: Action) {
+
+function pickAction(user: User, action: Action) 
+{
 	playerSelectedAction[user.id] = action;
 	gameServer.setPrivateData(user.id, 'selected-action', action);
 }
 
-function finalizeResults(players: User[]) {
-	// TODO: determine who won based on playerSelectedAction and increment their score
-	// TODO: if player hasn't picked an action then pick a random one for them
-	const playerWhoWon = players[randomIntFromInterval(0, players.length - 1)];
-	gameServer.setScore(playerWhoWon.id, gameServer.getScore(playerWhoWon.id) + 1);
 
-	movePlayerToResults(players[0], players[1], playerWhoWon.id === players[0].id ? 'won' : 'lost');
-	movePlayerToResults(players[1], players[0], playerWhoWon.id === players[1].id ? 'won' : 'lost');
+function finalizeResults(players: User[]) 
+{
+	// determine who won based on playerSelectedAction and increment their score
 
-	// TODO: Do another round if players had a tie
+	// if a player hasn't picked an action pick a random one for them
+	const actions = ['rock', 'paper', 'scissors'] as Action[]
 
-	// Short delay to show vistory/loss animation before we reset player states
-	setTimeout(() => {
-		// Resets selected player action
-		for (let i = 0; i < players.length; i += 1) {
-			pickAction(players[i], null);
-			movePlayerToMainMenu(players[i]);
-		}
-	}, 2000);
+	if (playerSelectedAction[players[0].id] == null)    pickAction(players[0], actions[Math.floor(Math.random() * actions.length)]);
```

Comment:
I would recommend using `randomIntFromInterval(0, actions.length - 1)` as it's much clearer to know what's going on instead of looking at multiple functions to get a random number :)